### ISSUE TITLE
"Upload Coverage Result To Coveralls" sometimes fails even when `fail_on_error: false` is set

### ISSUE THREAD

**Issue by patbl** [coverallsapp/orb#40]
I've had a bunch of build failures since starting to use the Coveralls orb. This is undesirable because it can block people from merging their changes for what we consider to be a non-essential part of the process. I thought that setting `fail_on_error: false` would prevent this, but it seems not to have.

Here's one of the errors I've encountered:

```
sha256sum: coveralls-linux.tar.gz: No such file or directory
coveralls-linux.tar.gz: FAILED open or read
sha256sum: WARNING: 1 listed file could not be read
tar (child): coveralls-linux.tar.gz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
Parsing args
Dry run - 0
Reporting coverage
+ ./coveralls report --no-fail
/bin/bash: line 74: ./coveralls: No such file or directory

Exited with code exit status 127
```

It appears to be failing on this line:

https://github.com/coverallsapp/orb/blob/b48e556b45f4119f82742212b9b75906bbdc9f9e/src/scripts/coveralls.sh#L75

I think this could be solved by adding `|| true` to the end of the command if `fail_on_error` is `false`.

I'm using `coveralls/coveralls@2.2.1`.

**Comment by afinetooth** [comment]
@patbl Thanks for reporting, and thanks so much for the potential fix. 

We aim to get that out next week.

**Comment by afinetooth** [comment]
Hi @patbl. We just released `v2.2.2` of the [coveralls orb](https://circleci.com/developer/orbs/orb/coveralls/coveralls), which fixes addresses this issue.

Thanks so much for your PR. We decided to make slightly more robust changes scoped to specific setup tasks that will be parallel to code changes in our [github-action](https://github.com/marketplace/actions/coveralls-github-action). (I will leave a note on your PR.)

I tested `fail_on_error` in a number of local tests with success:

1. Failure of download
2. Failure to verify checksums
3. Failure to find binary named `coveralls`

We didn't have time right now, but we do intend to implement those tests (of "setup failures") as new steps in our CI/CD pipeline at some point.

I would say "Let me know if it doesn't work for you," but I hope you won't encounter any more setup issues anytime soon.
